File: seminar/src/main/java/com/server/dosopt/seminar/common/config/BCryptPasswordConfig.java
package com.server.dosopt.seminar.common.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

@Configuration
public class BCryptPasswordConfig {

    private static final int STRENGTH = 10;   // default: 10 *많을수록 좋지만, 너무 많아져도 별 의미가 없어짐

    @Bean
    public PasswordEncoder bCryptPasswordEncoder() {
        return new BCryptPasswordEncoder(STRENGTH);
    }
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/common/config/BCryptPasswordConfig.kt
package sopt.org.kotlinseminar.common.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

private const val STRENGTH = 10

@Configuration
class BCryptPasswordConfig {

    @Bean
    fun bCryptPasswordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder(STRENGTH)
    }
}